src/renderers/shared/ReactDebugTool.js

```diff
@@ -52,6 +52,9 @@ var currentTimerStartTime = null;
 var currentTimerNestedFlushDuration = null;
 var currentTimerType = null;
 
+var beginLifeCycleTimerHasWarned = false;
```

Comment:
I think it’s fine to have a single variable for this. If it warns once, something’s messed up already so it might warn more times.